File: benchmarks/python/exponentialfcn.py
import numpy as np

def exponentialfcn(x: np.ndarray) -> np.ndarray:
    """
    Computes the value of the Exponential function.
    
    Parameters:
    -----------
    x : np.ndarray
        Input matrix of size M-by-N.
    
    Returns:
    --------
    scores : np.ndarray
        Output vector of size M-by-1 containing the function value for each
        row of x.
        
    Author: Mazhar Ansari Ardeh
    Please forward any comments or bug reports to mazhar.ansari.ardeh at
    Google's e-mail service or feel free to kindly modify the repository.
    """
    
    x2 = x ** 2
    scores = -np.exp(-0.5 * np.sum(x2, axis=1))
    return scores

File: src/matlab/exponentialfcn.m
% Computes the value of the Exponential function.
% SCORES = EXPONENTIALFCN(X) computes the value of the Exponential
% function at point X. EXPONENTIALFCN accepts a matrix of size M-by-N and
% returns a vetor SCORES of size M-by-1 in which each row contains the
% function value for the corresponding row of X.
% For more information, please visit:
% benchmarkfcns.info/doc/exponentialfcn
%
% Author: Mazhar Ansari Ardeh
% Please forward any comments or bug reports to mazhar.ansari.ardeh at
% Google's e-mail service or feel free to kindly modify the repository.
function scores = exponentialfcn(x)
   x2 = x .^2;

   scores = -exp(-0.5 * sum(x2, 2));
end
